sql/sql_admin.cc

```diff
@@ -1608,18 +1608,44 @@ bool Sql_cmd_optimize_table::execute(THD *thd)
                          FALSE, UINT_MAX, FALSE))
     goto error; /* purecov: inspected */
 
-  WSREP_TO_ISOLATION_BEGIN_WRTCHK(NULL, NULL, first_table);
+#ifdef WITH_WSREP
+  if (WSREP(thd) &&
+      (!thd->is_current_stmt_binlog_format_row() ||
+       !thd->find_temporary_table(first_table)))
+  {
+    /*
+      It makes sense to set auto_increment_* to defaults in TOI operations.
+      Must be done before wsrep_TOI_begin() since Query_log_event encapsulating
+      TOI statement and auto inc variables for wsrep replication is constructed
+      there. Variables are reset back in THD::reset_for_next_command() before
+      processing of next command.
+    */
+    if (wsrep_auto_increment_control)
+    {
+      thd->variables.auto_increment_offset = 1;
+      thd->variables.auto_increment_increment = 1;
+    }
+
+    wsrep::key_array keys;
+    if (!wsrep_append_fk_parent_table(thd, first_table, &keys))
```

Comment:
This call to function `wsrep_append_fk_parent_table()` will be executed also when TOI applying. This is unnecessary. 
I think it would be best to add clause `wsrep_thd_is_local(thd)` to the `if` statement above.

Reply:
Fixed as suggested.